server/sonar-server/src/main/java/org/sonar/server/computation/metric/MetricImpl.java

```diff
@@ -60,6 +61,13 @@ public MetricType getType() {
     return type;
   }
 
+  public boolean isNumericType() {
```

Comment:
This function returns `true` when valuetype is `BOOLEAN` which is not understandable to anyone. This function is more `isStoredAsADouble` or `isConvertibleToDouble`.

But Storage is : 
1. no concern to this Metric interface as a element of the Compute Engine. Abstracting away the storing concerns is the point of having interfaces and repositories in the CE
2. is related to the Measure and not to the Metric

In addition, this method is used in a single step. This is internal mechanic of this step. It should not be moved to the Metric interface (nor the Measure class).

Finally, it misses the `@Override` annotation